src/main/java/com/google/cloud/tools/maven/StageMojo.java

```diff
@@ -193,6 +196,11 @@ public void execute() throws MojoExecutionException, MojoFailureException {
     if (new File(sourceDirectory.toString() + "/WEB-INF/appengine-web.xml").exists()) {
       getLog().info("Detected App Engine standard environment application.");
 
+      // force runtime to 'java' for compat projects using Java version >1.7
+      if (Float.parseFloat(getCompileTargetVersion()) > 1.7f) {
```

Comment:
I do, but it seems more verbose. I don't think the Java target version can ever be anything but a real number.